examples/NAT.cs

```diff
@@ -36,223 +36,348 @@
 using PacketDotNet;
 using System.Net;
 using Pax;
+using System.Net.NetworkInformation;
 
 public class NAT : SimplePacketProcessor {
   public const int Port_Drop = -1;
   public const int Port_Outside = 0;
 
   private IPAddress my_address;
   private ushort next_port;
+  private readonly object portLock = new object();
+  private readonly PhysicalAddress next_hop_mac;
```

Comment:
This could be renamed to something like `next_outside_hop_mac` to make the orientation of "next" clearer.